core/src/main/scala/org/apache/spark/scheduler/JobWaiter.scala

```diff
@@ -17,6 +17,10 @@
 
 package org.apache.spark.scheduler
 
+import sun.misc.{Signal, SignalHandler}
```

Comment:
@hvanhovell I tried what you suggested [here](https://github.com/piaozhexiu/spark/commit/1cae5da93e44a82a6da7828203ce090e5afb1107), but that makes [jenkins build](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/41011/console) time out. Seems Java reflection adds overheads, so I had to revert it.